src/adagrams.js

```diff
@@ -1,8 +1,118 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
+class Adagrams {
+  
+  static LETTERS = {
+    'A': {num: 9, val: 1}, 
+    'B': {num: 2, val: 3}, 
+    'C': {num: 2, val: 3}, 
+    'D': {num: 4, val: 2}, 
+    'E': {num: 12, val: 1}, 
+    'F': {num: 2, val: 4}, 
+    'G': {num: 3, val: 2}, 
+    'H': {num: 2, val: 4}, 
+    'I': {num: 9, val: 1}, 
+    'J': {num: 1, val: 8}, 
+    'K': {num: 1, val: 5}, 
+    'L': {num: 4, val: 1}, 
+    'M': {num: 2, val: 3}, 
+    'N': {num: 6, val: 1}, 
+    'O': {num: 8, val: 1}, 
+    'P': {num: 2, val: 3}, 
+    'Q': {num: 1, val: 10}, 
+    'R': {num: 6, val: 1}, 
+    'S': {num: 4, val: 1}, 
+    'T': {num: 6, val: 1}, 
+    'U': {num: 4, val: 1}, 
+    'V': {num: 2, val: 4}, 
+    'W': {num: 2, val: 4}, 
+    'X': {num: 1, val: 8}, 
+    'Y': {num: 2, val: 4}, 
+    'Z': {num: 1, val: 10},
+  };
+  
+  static drawLetters() {
+    const letterPool = []; 
+    for (const char in this.LETTERS) {
+      for (let i=0; i<this.LETTERS[char]['num']; i++) letterPool.push(char);
+    }; // ensure a more weighted distribution
+
+    const drawn = [];
+    for (let i=0; i<10; i++) {
+      let char = letterPool[Math.floor(Math.random()*letterPool.length)]; // see Paul J.'s answer: https://stackoverflow.com/questions/2532218/pick-random-property-from-a-javascript-object
+
+      while (this.LETTERS[char]['num'] < 1) char = letterPool[Math.floor(Math.random()*letterPool.length)];
+
+      drawn.push(char);
+      this.LETTERS[char]['num']--;
+    };
+
+    return drawn;
+  };
+
+  static usesAvailableLetters(input, lettersInHand) {
+    const inputCharCount = {};
+    for (let i=0; i<input.length; i++) {
+      (inputCharCount[input[i]]) ? inputCharCount[input[i]]++ : inputCharCount[input[i]] = 1;  
+    }; // create hashmap for input
```

Comment:
This is mostly a preference of mine, but I prefer to use `forEach` or `for-in` loops when traversing a string.  There's less of a chance of typos since the for loop has those three sections it's easy to make a typo.